### PR TITLE
1226 1229 combine dashboard form patterns

### LINKED ISSUE #1226: Combine dashboard theme forms base

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Ports over the new form layout styles from dashboard
- Creates example and docs

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- View new form examples at http://0.0.0.0:8101/vanilla-framework/examples/patterns/forms/forms/

## Details

Fixed #1226 #1229

### CODE REVIEW

**Comment by kwm14** [comment]
## Design review

- [ ] Add `1rem` of spacing in between 'Choose file' button and text
- [ ] Text labels should use a max width of `3-col`
- [ ] We have a date dropdown inside a email address form - is that right?

**Comment by kwm14** [comment]
Ignore my previous review as I was looking at forms-base rather than forms-forms. Will re-review now.

**Comment by kwm14** [comment]
Looking good :+1:

File: docs/en/patterns/forms.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Forms layout
+table_of_contents: true
+---
+
+# Form inline and stacked styles
+
+By applying the classes ```.p-form--inline .p-form--stacked``` and wrapping any form control in ```.p-form__group``` you can change the layout style of any form and to fit a design purpose.
+
+<a href="https://vanilla-framework.github.io/vanilla-framework/examples/patterns/forms/"
```
Comment (anthonydillon) [inline docs/en/patterns/forms.md]: Again this link should be forms/form but the doc shouldn't be in a directory on its own.

File: examples/patterns/forms/forms.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    <div class="p-form__control">
+      <input type="text" id="address-optional-stacked">
+    </div>
+  </div>
+  <button class="p-button--positive u-float-right">Add details</button>
+</form>
+<form class="p-form p-form--inline u-float-left">
+  <div class="p-form__group">
+    <label for="full-name-inline" class="p-form__label">Full name</label>
+    <input type="text" id="full-name-inline" class="p-form__control" required>
```
Comment (anthonydillon) [inline examples/patterns/forms/forms.html]: Same ID used.

File: examples/patterns/forms/forms.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  </div>
+  <div class="p-form__group">
+    <label for="address-optional-stacked" class="p-form__label">Address line 2 (optional)</label>
+    <div class="p-form__control">
+      <input type="text" id="address-optional-stacked">
+    </div>
+  </div>
+  <button class="p-button--positive u-float-right">Add details</button>
+</form>
+<form class="p-form p-form--inline u-float-left">
```
Comment (anthonydillon) [inline examples/patterns/forms/forms.html]: This should be in its own example and the docs should just have two examples with documentation.

File: scss/_patterns_forms.scss [hunk 4]
```diff
@@ -0,0 +1,103 @@
+@import 'settings';
+
+@mixin vf-p-forms {
+  .p-form {
+
+    * + &__control {
```
Comment (anthonydillon) [inline scss/_patterns_forms.scss]: Why does this need *?

File: examples/patterns/forms/forms.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
+layout: default
+title: Forms / Layout
+category: _patterns
+---
+
+<form class="p-form p-form--stacked u-float-left">
```
Comment (anthonydillon) [inline examples/patterns/forms/forms.html]: We should choose if inline our stacked is default and make that just `p-form` and then create a single modifier class for the other.

**Comment by anthonydillon** [comment]
Also add to docs navigation.

File: examples/patterns/forms/forms.html (hunk 5 above)
Comment (richmccartney) [inline examples/patterns/forms/forms.html]: Ah see I thought the same but you can also just do a forms without classes so this doesn't really apply. Also there are quite a few discussions about when you should change form layout or not. Best to keep it as is for the time being.

File: examples/patterns/forms/forms.html (hunk 3 above)
Comment (richmccartney) [inline examples/patterns/forms/forms.html]: Good call, I'll split them.

File: scss/_patterns_forms.scss (hunk 4 above)
Comment (richmccartney) [inline scss/_patterns_forms.scss]: I have no clue 🤔 I've removed it.

File: examples/patterns/forms/forms.html (hunk 2 above)
Comment (richmccartney) [inline examples/patterns/forms/forms.html]: Thats a `for` and an `id` dude 😄

**Pull request merged**